File: 10845/10845.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10845.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/03/01 16:36:19 by jihoolee          #+#    #+#             */
/*   Updated: 2024/03/01 16:51:14 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <string>
#include <queue>

int front_queue(std::queue<int>* p_q) {
  if (p_q->empty())
    return -1;
  return p_q->front();
}

int back_queue(std::queue<int>* p_q) {
  if (p_q->empty())
    return -1;
  return p_q->back();
}

int pop_queue(std::queue<int>* p_q) {
  if (p_q->empty())
    return -1;

  int popped = p_q->front();

  p_q->pop();
  return popped;
}

void do_cmd(const std::string& cmd, std::queue<int>* p_q) {
  if (cmd == "push") {
    int num;
    std::cin >> num;
    p_q->push(num);
  } else if (cmd == "pop") {
    std::cout << pop_queue(p_q) << std::endl;
  } else if (cmd == "size") {
    std::cout << p_q->size() << std::endl;
  } else if (cmd == "empty") {
    std::cout << int(p_q->empty()) << std::endl;
  } else if (cmd == "front") {
    std::cout << front_queue(p_q) << std::endl;
  } else if (cmd == "back") {
    std::cout << back_queue(p_q) << std::endl;
  }
}

int main(void) {
  int             cmd_cnt;
  std::string     cmd;
  std::queue<int> q;

  std::cin >> cmd_cnt;
  for (int i = 0; i < cmd_cnt; ++i) {
    std::cin >> cmd;
    do_cmd(cmd, &q);
  }
  return 0;
}

File: 10845/10845.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10845.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/03/01 16:53:41 by jihoolee          #+#    #+#              #
#    Updated: 2024/03/01 17:24:26 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys
from collections import deque

input = sys.stdin.readline


class Queue:
    def __init__(self):
        self.container: deque[int] = deque()

    def push(self, num: int) -> None:
        self.container.append(num)

    def pop(self) -> int:
        if (self.empty()):
            return -1
        return self.container.popleft()

    def size(self) -> int:
        return len(self.container)

    def empty(self) -> bool:
        return len(self.container) == 0

    def front(self) -> int:
        if (self.size() == 0):
            return -1
        return self.container[0]

    def back(self) -> int:
        if (self.size() == 0):
            return -1
        return self.container[-1]


def main() -> None:
    q: Queue = Queue()

    for _ in range(int(input())):
        cmd: list[str] = input().split()

        if cmd[0] == 'push':
            q.push(int(cmd[1]))
        elif cmd[0] == 'pop':
            print(q.pop())
        elif cmd[0] == 'size':
            print(q.size())
        elif cmd[0] == 'empty':
            print(int(q.empty()))
        elif cmd[0] == 'front':
            print(q.front())
        elif cmd[0] == 'back':
            print(q.back())


if __name__ == '__main__':
    main()
